src/rebdhuhn/graphviz.py

```diff
@@ -143,15 +162,103 @@ def _convert_node_to_dot(ebd_graph: EbdGraph, node: str, indent: str) -> str:
             raise ValueError(f"Unknown node type: {ebd_graph.graph.nodes[node]['node']}")
 
 
+def _get_multi_step_instruction_node_key(instruction: MultiStepInstruction) -> str:
+    """
+    Returns the node key for a multi-step instruction node.
+    Format: msi_{first_step_number_affected}
+    """
+    return f"msi_{instruction.first_step_number_affected}"
+
+
+def _convert_multi_step_instruction_to_dot(instruction: MultiStepInstruction, indent: str) -> str:
+    """
+    Convert a MultiStepInstruction to a dot node.
+    Uses light blue background (#e6f3ff) to distinguish from outcome notes.
+    """
+    # Format the instruction text with word wrapping
+    label_with_linebreaks = add_line_breaks(
+        instruction.instruction_text, max_line_length=_MSI_LABEL_MAX_LINE_LENGTH, line_sep="\n"
+    )
+    formatted_label = escape(label_with_linebreaks).replace("\n", '<BR align="left"/>')
+    formatted_label = f'<FONT><I>{formatted_label}</I></FONT><BR align="left"/>'
+
+    node_key = _get_multi_step_instruction_node_key(instruction)
+    # pylint:disable=line-too-long
+    return (
+        f'{indent}"{node_key}" '
+        f'[margin="0.2,0.12", shape=note, style=filled, penwidth=0.0, fillcolor="{_MSI_NODE_BGCOLOR}", '
+        f'label=<{formatted_label}>, fontname="Roboto, sans-serif"];'
+    )
+
+
+def _collect_node_keys_in_step_range(ebd_graph: EbdGraph, start_step: str, end_step: str | None) -> list[str]:
```

Comment:
why is the result a list here? does order matter? asking because other methods return node keys as set.